bene_promo_redirect.module

```diff
@@ -13,38 +13,38 @@ use Drupal\Component\Utility\UrlHelper;
 function bene_promo_redirect_page_attachments(array &$page) {
   $config = \Drupal::config('bene_promo_redirect.settings');
   // Only redirect when enabled:
-  if ($config->get('enabled')) {
-    $destination = $config->get('dest');
-    if (!empty($destination) && UrlHelper::isValid($destination)) {
-      $route = \Drupal::routeMatch()->getRouteObject();
-      // Don't redirect admin pages:
-      if (!\Drupal::service('router.admin_context')->isAdminRoute($route)) {
-        $request = \Drupal::request();
-        $current_uri = $request->getRequestUri();
-        // Don't redirect User pages:
-        if (strpos($current_uri, "/user") !== 0) {
-          // Now let's match against our path specifications:
-          if ($config->get('redirect_all')) {
-            $pathmatch = TRUE;
-          }
-          else {
-            $pathmatch = FALSE;
-            foreach ($config->get('paths') as $path) {
-              if (in_array($current_uri, [$path, "/$path"])) {
-                $pathmatch = TRUE;
-                continue;
-              }
+  $page['#cache']['tags'][] = 'config:bene_promo_redirect.settings';
+  $destination = $config->get('dest');
+  if (!empty($destination) && UrlHelper::isValid($destination)) {
+    $route = \Drupal::routeMatch()->getRouteObject();
+    // Don't redirect admin pages:
+    if (!\Drupal::service('router.admin_context')->isAdminRoute($route)) {
+      $request = \Drupal::request();
+      $current_uri = $request->getRequestUri();
+      // Don't redirect User pages:
+      if (strpos($current_uri, "/user") !== 0) {
+        // Now let's match against our path specifications:
+        if ($config->get('redirect_all')) {
+          $pathmatch = TRUE;
+        }
+        else {
+          $pathmatch = FALSE;
+          foreach ($config->get('paths') as $path) {
+            if (in_array($current_uri, [$path, "/$path"])) {
+              $pathmatch = TRUE;
+              continue;
             }
           }
-          if ($pathmatch) {
+        }
+        if ($pathmatch) {
+          if ($config->get('enabled')) {
```

Comment:
@mariacha is there a reason to move this? It seems like the easiest check to run first, since we are moving the cache tag all the way to the top of this function anyway. I feel like I'm missing something.

Reply:
Naw, I think I moved this here first, then didn't un-move it when it didn't fully work.